features/FxA-83-sign-in-confirmation-email/README.md

```diff
@@ -0,0 +1,173 @@
+
+* The solution must work across all clients,
+  without patches landing in client code.
+
+* The solution must not affect existing sessions
+  from the user's point of view.
+
+* The solution must work from
+  regular Sync sign-in,
+  `/force_auth` and
+  when signing-in to Sync from `/signup`.
+
+## Proposed solution
+
+Add a notion of “verified”
+to sessionTokens and keyFetchTokens.
+Tokens that are unverified have reduced powers,
+meaning that requests to protected endpoints will fail
+with error number 102 (unverified user).
+Token verification is achieved
+by following a link sent by email.
+
+## User stories
+
+* As a Sync user,
+  when signing in to my Firefox Account,
+  I want my identity confirmed
+  via my email account.
+
+* As a user of an OAuth relier,
+  when signing in to my Firefox Account,
+  I do not want an extra confirmation step.
+
+## Work breakdown
+
+### fxa-content-server
+
+- [ ] Add strings to `strings.js`
+  to get localization done quicker.
+  This task is the highest immediate priority.
+- [ ] Implement "confirm your email" screen.
+- [ ] Implement confirmation landing screen.
+- [ ] Add handling for `verified` and `challenge` fields
+  in `/account/login` response.
+- [ ] Add handling for 102 errors
+  from `/certificate/sign`
+  and `/account/keys` endpoints.
+
+### fxa-auth-server
+
+- [ ] Add method to `mailer.js`
+  for sending confirmation email.
+- [ ] Modify `/account/login`
+  to send verification status to db,
+  initiate verification email
+  and set `challenge` on response.
+- [ ] Create a `verifiedSessionToken`
+  auth strategy for Hapi
+  that fails with 102 (unverified user) error
+  if sessionToken is not verified.
+- [ ] Create a `verifiedKeyFetchToken`
+  auth strategy for Hapi
+  that fails with 102 (unverified user) error
+  if keyFetchToken is not verified.
+- [ ] In `/certificate/sign`,
+  use new `verifiedSessionToken` auth strategy.
+- [ ] In `/account/keys`,
+  use new `verifiedKeyFetchToken` auth strategy.
+- [ ] Add a new endpoint for token verification.
+
+### fxa-auth-db-mysql
+
+- [ ] Create `tokenVerifications` table
+  and associated stored procedures.
+- [ ] Update token-creation stored procedures
+  to also insert into `tokenVerifications`
+  as part of the same transaction.
+- [ ] Update `PUT /sessionToken/:id` endpoint
+  to accept verification state.
+- [ ] Update `PUT /keyFetchToken/:id` endpoint
+  to always create token unverified.
+- [ ] Add `/token/:id/verify` endpoint for verifying tokens.
+- [ ] Add stored procedure and endpoint
+  that returns token joined to its `tokenVerifications` row
+  for callers that need to check token verification state.
+
+### fxa-auth-mailer
+
+- [ ] Add confirmation email templates and methods
+
+## Mock-ups
+
+### Confirm this sign-in screen
+
+![Mock-up of the "Confirm this sign-in" screen](confirm-this-sign-in.png)
+
+### Confirmation email
+
+![Mock-up of the confirmation email](confirmation-email.png)
```

Comment:
@ryanfeeley - at the bottom of the email, there is text that says "If you did not authorize this action, then please reset your password to erase your data from our servers." Is this a copy-paste remnant? This feature forces a re-confirmation before an unauthorized person can access that data. 

It seems like this text should be removed or replaced with something along the lines of "Check for follow on emails that indicates whether someone else has been accessed your account. If so, __something actionable here__"